### PR TITLE
Addition of a new SDACT with up and down command

### PR DESCRIPTION
**Pull request by ghairapetian**

### CODE REVIEW

File: ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml [hunk 1]
```diff
@@ -6791,6 +6791,15 @@ OAQM:
...
   - MONITORING
 
+UDEC:
+  description: "Up and down extension control for shades."
+  is_abstract: true
+  uses:
+  - up_shade_command
+  - down_shade_command
+  opt_uses:
+  - shade_extent_percentage_sensor
```
Comment (cstirdivant) [inline ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml:6801]: Does this device have `shade_extent_percentage_command` ? It seems odd to have the up/down commands without a position command; for example, `shade_extent_percentage_command` is essentially the position setpoint that would tell the up/down commands when to start or stop.

**Review by cstirdivant — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml (hunk 1 above)
Comment (ghairapetian) [reply ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml:6801]: I agree with you it's odd. However, there is no position command. The only way to know when starting and stopping the up and down command it's by calculating the duration of the command according to the current position and the shade speed. For exemple, if the controller want to go from 80% to 20% and it's known the shade go up at the speed of 10% per second, the up command will be apply 6 seconds.

Comment (cstirdivant) [reply ontology/yaml/resources/HVAC/entity_types/ABSTRACT.yaml:6801]: Ok thank you for clarifying.

**Review by cstirdivant — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 2]
```diff
@@ -3175,10 +3175,18 @@ literals:
   - ACTIVE
   - INACTIVE
 
+- up_shade_command:
+  - ACTIVE
```
[the remark this answers is not in the document]
Comment (ghairapetian) [reply ontology/yaml/resources/fields/telemetry_fields.yaml]: Done

File: ontology/yaml/resources/subfields/subfields.yaml [hunk 3]
```diff
@@ -242,6 +243,7 @@ descriptor:
   transducer: "A device that converts energy from one form to another (e.g., sound energy to electrical energy in a microphone, electrical energy to sound energy in a speaker, etc.)."
   ultraviolet: "A short wavelength type of light that is beyond the visible spectrum. It is damaging to DNA and cells, and is commonly used to sterilize surfaces."
   unoccupied: "State of being outside of scheduled (time of day or week) run-time parameters."
+  up: "The direction of a spatial movement, opposite of down."
```
Comment (cstirdivant) [inline ontology/yaml/resources/subfields/subfields.yaml:246]: Can you please move these fields (down, up) to the `measurement_descriptor` section (they are similar to open/closed)

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 4]
```diff
@@ -3175,10 +3175,18 @@ literals:
   - ACTIVE
   - INACTIVE
 
+- up_shade_command:
```
Comment (cstirdivant) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:3178]: Re-name to `shade_up_command` after moving the subfields to the `measurement_descriptor` category

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 5]
```diff
@@ -3175,10 +3175,18 @@ literals:
...
+  - ON
+  - OFF
+- down_shade_command:
```
Comment (cstirdivant) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:3181]: Re-name to `shade_down_command` after moving the subfields to the `measurement_descriptor` category

**Review by cstirdivant — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/subfields/subfields.yaml (hunk 3 above)
Comment (ghairapetian) [reply ontology/yaml/resources/subfields/subfields.yaml]: done

File: ontology/yaml/resources/fields/telemetry_fields.yaml (hunk 4 above)
Comment (ghairapetian) [reply ontology/yaml/resources/fields/telemetry_fields.yaml]: done

File: ontology/yaml/resources/fields/telemetry_fields.yaml (hunk 5 above)
Comment (ghairapetian) [reply ontology/yaml/resources/fields/telemetry_fields.yaml]: done

**Review by cstirdivant — APPROVED** [review]

**Pull request merged**